getanewsletter.php

```diff
@@ -212,34 +221,39 @@ function news_js_ajax()
 	$pluginurl = WP_PLUGIN_URL.'/'.str_replace(basename( __FILE__),"",plugin_basename(__FILE__));
 	?>
 	<script type="text/javascript">
-	//<![CDATA[
-	jQuery(document).ready(function()
-	{
-		jQuery(".news-loading").hide();
-		jQuery(".newsletter-signup").submit(function(){
-			var str = jQuery(this).serialize();
-
-			jQuery.ajax({
-				type: "POST",
-				url: "<?php echo $pluginurl; ?>subscribe.php",
-				data: str,
-				beforeSend: function (msg){
-				  jQuery(".news-loading").show();
-				},
-				success: function(msg){
-					jQuery(".news-note").ajaxComplete(function(event, request, settings){
-						jQuery(this).html(msg);
-						if(msg == '<span class="news-success"><?php echo get_option("newsletter_msg_success"); ?></span>' || msg == '<span class="news-success"><?php echo get_option("newsletter_msg_confirm"); ?></span>') 
-							jQuery(".newsletter-signup").hide();
-						jQuery(".news-loading").hide();
-					});
-				}
-			});
-		return false;
-		});
-	});
-	//]]>
+    	//<![CDATA[
+    	jQuery(document).ready(function()
+    	{
+    		jQuery('.news-loading').hide();
+
+    		jQuery('.newsletter-signup').submit(function() {
+    			var data = jQuery(this).serialize();
+
+    			jQuery.ajax({
+    				type: 'POST',
+    				url: '<?php echo admin_url('admin-ajax.php'); ?>',
+    				data: {
+						action: 'getanewsletter_subscribe',
+						data: data
```

Comment:
this part doesn't work, I think you need to mapp $_POST['id_name'] to $_POST['data']['id_name'] since you changed the data
https://github.com/akochetyga/wp-get-a-newsletter/blob/develop/subscribe.php